crates/agglayer-certificate-orchestrator/src/lib.rs

```diff
@@ -279,6 +256,100 @@ where
         Ok(())
     }
 
+    /// Function that handles the end of an epoch.
+    /// This function is called when the orchestrator receives an EpochEnded
+    /// event. The function is responsible for:
+    /// - Opening the next epoch.
+    /// - Spawning the epoch packing task.
+    fn handle_epoch_end(&mut self, epoch: u64) -> Result<(), Error> {
+        debug!("Start the settlement of the epoch {}", epoch);
+        let task = self.epoch_packing_task_builder.clone();
+
+        let closing_epoch = self.current_epoch.load_full();
+        // TODO: Check for overflow
+        let next_epoch = epoch + 1;
+
+        match self.epochs_store.open(next_epoch) {
+            Ok(new_epoch) => self.current_epoch.store(Arc::new(new_epoch)),
+            Err(error) => {
+                error!(
+                    "CRITICAL error: Failed to open the next epoch {}: {:?}",
+                    next_epoch, error
+                );
+
+                return Err(Error::InternalError);
+            }
+        }
+
+        self.epoch_packing_tasks
+            .spawn(async move { task.pack(closing_epoch)?.await });
+
+        Ok(())
+    }
+}
+
+// This block contains the logic applied to a Certificate.
+// The method are executed following this flow:
+//
+//                      ┌─────────────────────────┐
+//                      │                         │
+// ┌────────────────────►  spawn certifier task   ◄───────────────────┐
+// │                    │                         │                   │
```

Comment:
nit: bit broken in terms of spaces it seems, also do we want to have that as rustdoc instead?

Reply:
This is applied to an `impl` which gather multiple methods, I don't think it renders on rustdoc

> Broken in terms of spaces
You mean that the diagram is too big?

Reply:
> You mean that the diagram is too big?

meant that they are some shift as it is, the rectangles and arrows are bit broken but i'm fine with it, I can still understand

Reply:
nevermind, the wrong rendering on my side seemed to be due to some font glitch